streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -474,12 +596,18 @@ private void cleanRemovedTasksCalledByCleanerThread(final long cleanupDelayMs) {
         for (final TaskDirectory taskDir : listAllTaskDirectories()) {
             final String dirName = taskDir.file().getName();
             final TaskId id = parseTaskDirectoryName(dirName, taskDir.namedTopology());
-            if (!lockedTasksToOwner.containsKey(id)) {
+            if (!lockedTasksToOwner.containsKey(id) || tasksForLocalState.containsKey(id)) {
```

Comment:
Oh yes, you're right. Actually, I don't think we should add anything to the condition here, and it should remain as just `if (!lockedTasksToOwner.containsKey(id))`, because we still want to close and delete _stale_ tasks, even if they're currently a "pending" task.